sopel/plugins/rules.py

```diff
@@ -1141,53 +1147,59 @@ def is_threaded(self):
-
-    def is_channel_rate_limited(self, channel):
-        metrics: RuleMetrics = self._metrics_sender.get(channel, RuleMetrics())
-        now = datetime.datetime.utcnow()
-        rate_limit = datetime.timedelta(seconds=self._channel_rate_limit)
-        return metrics.is_limited(now - rate_limit)
-
-    def is_global_rate_limited(self):
-        now = datetime.datetime.utcnow()
-        rate_limit = datetime.timedelta(seconds=self._global_rate_limit)
-        return self._metrics_global.is_limited(now - rate_limit)
+    def get_user_metrics(self, nick) -> RuleMetrics:
+        return self._metrics_nick.get(nick, RuleMetrics())
 
-    def get_user_rate_message(self, nick):
-        template = self._user_rate_message or self._default_rate_message
+    def get_channel_metrics(self, channel) -> RuleMetrics:
+        return self._metrics_sender.get(channel, RuleMetrics())
 
-        if not template:
-            return None
+    def get_global_metrics(self) -> RuleMetrics:
+        return self._metrics_global
 
-        return template.format(
-            nick=nick,
-        )
-
-    def get_channel_rate_message(self, nick, channel):
-        template = self._channel_rate_message or self._default_rate_message
+    @property
+    def user_rate_limit(self) -> datetime.timedelta:
+        return datetime.timedelta(seconds=self._user_rate_limit)
 
-        if not template:
-            return None
+    @property
+    def channel_rate_limit(self) -> datetime.timedelta:
+        return datetime.timedelta(seconds=self._channel_rate_limit)
 
-        return template.format(
-            nick=nick,
-            channel=channel,
-        )
+    @property
+    def global_rate_limit(self) -> datetime.timedelta:
+        return datetime.timedelta(seconds=self._global_rate_limit)
+
+    def is_user_rate_limited(self, nick, at_time=None) -> bool:
+        if at_time is None:
+            at_time = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc)
+        metrics = self.get_user_metrics(nick)
+        return metrics.is_limited(at_time - self.user_rate_limit)
+
+    def is_channel_rate_limited(self, channel, at_time=None) -> bool:
+        if at_time is None:
+            at_time = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc)
+        metrics = self.get_channel_metrics(channel)
+        return metrics.is_limited(at_time - self.channel_rate_limit)
+
+    def is_global_rate_limited(self, at_time=None) -> bool:
+        if at_time is None:
+            at_time = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc)
+        metrics = self.get_global_metrics()
+        return metrics.is_limited(at_time - self.global_rate_limit)
 
-    def get_global_rate_message(self, nick):
-        template = self._global_rate_message or self._default_rate_message
+    @property
+    def user_rate_template(self):
+        template = self._user_rate_message or self._default_rate_message or None
```

Comment:
`self._default_rate_message` should be `Optional[str]` anyway, so I don't think the `or None` is necessary:

```suggestion
        template = self._user_rate_message or self._default_rate_message
```